octodns/record/__init__.py

```diff
@@ -304,7 +304,14 @@ def __init__(self, zone, name, data, source=None):
         self.values = sorted(self._value_type.process(values))
 
     def changes(self, other, target):
-        if self.values != other.values:
+        self_values = self.values
+        other_values = other.values
+        # compare IPv6 addresses separately since they could be
+        # in short/long format
+        if self._type == 'AAAA':
+            self_values = [IPv6Address(x) for x in self_values]
+            other_values = [IPv6Address(x) for x in other_values]
+        if self_values != other_values:
```

Comment:
Good catch!

Looks like we already convert things to `IPv6Address`/`IPv4Address` as part of validation. Shame we didn't go ahead and make the values IP types. 

This solution will work, but I'll probably poke around a bit and see if I can come up with a way that avoids having `_ValuesMixin` know about the details of some of it's potential types.

Reply:
After beating my head on this for a while I realized the most fullproof solution was to normalize the representation using IPv[46]Address during the record create. 

Looking back I wish I'd have the values be the actual types and not just strings, but that'd be too big of a shift now...

Normalization PR is https://github.com/octodns/octodns/pull/759. Thanks for reporting this one.